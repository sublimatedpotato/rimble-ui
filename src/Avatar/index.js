import React from 'react';
import styled from 'styled-components';
import { themeGet } from 'styled-system';

import Box from '../Box';

import defaultTheme from '../theme';

const size = props => {
  switch (props.size) {
    case 'small':
      return `
        height: 2em;
        width: 2em;
      `;
    case 'medium':
      return `
        height: 3em;
        width: 3em;
      `;
    case 'large':
      return `
        height: 4em;
        width: 4em;
      `;
    default:
      return ``;
  }
};

const Avatar = styled(Box)`
  & {
    position: relative;
    overflow: hidden;
    border-radius: 100%;
    background: center no-repeat;
    background-size: cover;
    background-color: ${props => themeGet('colors.grey', '#000')};
    background-image: url(${props => props.src});
  }

  ${size}
`;

Avatar.defaultProps = {
  theme: defaultTheme,
  size: '2rem',
  alt: 'Avatar',
};

Avatar.displayName = 'Avatar';

export default Avatar;
