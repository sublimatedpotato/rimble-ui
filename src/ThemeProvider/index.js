import React from 'react';
import PropTypes from 'prop-types';

import styled, {
  createGlobalStyle,
  ThemeProvider as StyledThemeProvider,
} from 'styled-components';

import { normalize, transparentize } from 'polished';

import theme from '../theme';

export const Base = styled.div`
  & {
    font-size: 1em;
    font-family: ${props => props.theme.fonts.sansSerif};
    color: ${props => props.theme.colors.copyColor};
  }
`;

const GlobalStyle = createGlobalStyle`
  ${normalize()}
  html {
    box-sizing: border-box;
  }
  *, *:before, *:after {
    box-sizing: inherit;
  }
  ::selection {
    background: ${props => transparentize(0.7, props.theme.colors.primary)};
  }
`;

const ThemeProvider = ({ ...props }) => {
  return (
    <StyledThemeProvider theme={theme} {...props}>
      <React.Fragment>
        <GlobalStyle />
        <Base {...props} />
      </React.Fragment>
    </StyledThemeProvider>
  );
};

ThemeProvider.propTypes = {};

export default ThemeProvider;
