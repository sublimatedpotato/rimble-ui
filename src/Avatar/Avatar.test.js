import Avatar from './';
import ThemeProvider from '../ThemeProvider';
import theme from '../theme';

describe('Avatar component sanity', () => {
  it('has name', () => {
    expect(Avatar.displayName).toBe('Avatar');
  });

  it('matches default snapshot', () => {
    const component = render(<Avatar />);
    expect(component).toMatchSnapshot();
  });

  it('matches themed snapshot', () => {
    const component = render(
      <ThemeProvider theme={theme}>
        <Avatar />
      </ThemeProvider>
    );
    expect(component).toMatchSnapshot();
  });
});
