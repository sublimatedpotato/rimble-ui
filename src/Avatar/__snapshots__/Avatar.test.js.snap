// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar component sanity matches default snapshot 1`] = `
.c0 {
  width: 2rem;
  height: 2rem;
}

.c0 {
  box-sizing: border-box;
}

.c0 {
  position: relative;
  overflow: hidden;
  border-radius: 100%;
  background: center no-repeat;
  background-size: cover;
  background-color: #CCC;
  background-image: url();
}

<div
  alt="Avatar"
  class="c0"
/>
`;

exports[`Avatar component sanity matches themed snapshot 1`] = `
.c1 {
  width: 2rem;
  height: 2rem;
}

.c1 {
  box-sizing: border-box;
}

.c1 {
  position: relative;
  overflow: hidden;
  border-radius: 100%;
  background: center no-repeat;
  background-size: cover;
  background-color: #CCC;
  background-image: url();
}

.c0 {
  font-size: 1em;
  font-family: "Source Sans Pro",-apple-system,sans-serif;
  color: #3F3D4B;
}

<div
  class="c0"
>
  <div
    alt="Avatar"
    class="c1"
  />
</div>
`;
